spec/lib/payrix/resource/base_spec.rb

```diff
@@ -0,0 +1,20 @@
+RSpec.describe Payrix::Resource::Base do
+  Payrix::Resource::Tests = Class.new(Payrix::Resource::Base) do
+    def initialize(params)
+      super(params, [:status, :amount])
+
+      @resource_name = 'test'
+    end
+  end
+
+  let(:resource) { Payrix::Resource::Tests.new(status: '1', amount: '500') }
+
+  it 'raises Payrix::Exceptions::ApiError with the message which includes the errors payload' do
+    expect { resource.create }
```

Comment:
Also noticed that this triggers a Faraday warning, seems the fix is to update the gem according to this [StackOverflow comment](https://stackoverflow.com/a/72525024/7304530). Perhaps we want to update the gem later. 

```ruby
Top level ::CompositeIO is deprecated, require 'multipart/post' and use `Multipart::Post::CompositeReadIO` instead!
Top level ::Parts is deprecated, require 'multipart/post' and use `Multipart::Post::Parts` instead!
```

Reply:
Good callout, can we add a GitHub issue for that?

Reply:
Cool, added #5.